en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,223 @@
+Enable the settings: `Blake2s hash algorithm`, `Entropy pool and strong random number generator`, and `Use interrupts to feed timing randomness to entropy pool`.
+
+![Menuconfig Crypto Options Set](../../assets/hardware/kconfig-crypto-2.png)
+
+::: tip
+Some of these options can be tweaked if desired.
+:::
+
+After enabling encryption settings, exit `menuconfig`.
+You can now build and test.
+
+## Download & Decrypt Log Files
+
+Encrypted log files are downloaded using the QGroundControl [Log Download](https://docs.qgroundcontrol.com/master/en/qgc-user-guide/analyze_view/log_download.html) view (**Analyze Tools > Log Download**) just like ordinary log files.
+The only difference is that for each flight you will need to download both the encrypted log file, and the file containing the encrypted symmetric key.
+
+The encrypted log file and encrypted symmetric key file are displayed with a timestamp (but no filename) in QGroundControl, as shown below.
+You can determine which files are associated based on their timestamps.
+
+![QGroundControl ULog Download](../../assets/qgc/analyze/encrypted_log.png)
+
+Select and download both files.
+
+Note that both files will be downloaded with the `.ulg` suffix.
+You can identify the symmetric key file, as it is usually much smaller than the log file (about 300 bytes)
+
+For convenience in the decryption step, you might rename the file extensions to add back the `.ulgc` (log) and `.ulgk` (key) file extensions.
+
+### Decrypt ULogs
+
+Before you can analyze your encrypted logs, you will need to decrypt them.
+There is a Python script that can be used to decrypt logs in `Tools/decrypt_ulog.py`.
+
+`decrypt_ulog.py` takes 3 arguments:
+
+1. The encrypted `.ulogc` file.
+2. The symmetric key `.ulogk` file.
+3. The decryption key (the RSA2048 `.pem` private key which is used to unwrap the `.ulogk` file).
+
+```sh
+usage: decrypt_ulog.py [-h] [ulog_file] [ulog_key] [rsa_key]
+
+CLI tool to decrypt an ulog file
+
+positional arguments:
+  ulog_file   .ulog file
+  ulog_key    .ulogk, encrypted key
+  rsa_key     .pem format key for decrypting the ulog key
+
+optional arguments:
+  -h, --help  show this help message and exit
+
+```
+
+As an example:
+
+```sh
+python3 decrypt_ulog.py \
+/home/john/Downloads/log_24_2024-10-6-23-39-50.ulgc \
+/home/john/Downloads/log_23_2024-10-6-23-39-48.ulgk \
+new_keys/private_key.pem
+```
+
+On success the decrypted log file is created with the `.ul` suffix instead of `.ulg`.
+Rename the file back to `.ulg` and it is now ready for flight review.
+
+## Generate RSA Public & Private Keys
+
+In a production environment, you should your own generated keys rather than using the example keys.
+
+To generate the `CONFIG_PUBLIC_KEY0` public key, you can run the `Tools/cryptotools.py` Python program.
+For example, this command generates files called `new.pub` and `new.json` file in `Tools/test_keys`:
+
+```sh
+python3 Tools/cryptotools.py  --genkey Tools/test_keys/new
+```
+
```

Comment:
```suggestion
You can now reference the path to this public exchange key in the `CONFIG_PUBLIC_KEY` keystore on the .px4board file. 
By default, the exchange key is associated with `CONFIG_PUBLIC_KEY0`, but this can be modified with the [SDLOG_EXCH_KEY](../advanced_config/parameter_reference.md#SDLOG_EXCH_KEY) parameter.
```